lib/cbus_elixir/pagination.ex

```diff
@@ -0,0 +1,61 @@
+defmodule CbusElixir.Pagination do
+  import Ecto.Query
+  alias CbusElixir.Repo
+
+  def paginate(query, page, per_page) when is_binary(page) do
+    paginate(query, String.to_integer(page), per_page)
+  end
+
+  def paginate(query, page, per_page) do
+    count = total_entries(query) |> Repo.one()
+    
+
+    paged_results = paged_query(query, page, per_page) |> Repo.all()
+    {results, has_next} = split_results(per_page, paged_results)
```

Comment:
At first glance, I was confused about the calculation of `has_next`, but it does seem like preloading (i+1) and checking to see if it's there is pretty slick